### PR TITLE
#19 -- POC, loop over all hosts.

### PR DESCRIPTION
**Pull request by apollo13**
The PR is mostly so Travis can run through it

### CODE REVIEW

**Comment by jezdez** [comment]
I think we should make that feature opt-out. It's just to big of a departure of the current system and I'm afraid of hard to debug errors that may be swallowed up by Django's reversal system. Maybe introduce a `HOST_STICKY` or something?

File: django_hosts/resolvers.py [hunk 1]
```diff
@@ -172,16 +172,31 @@ def reverse(viewname, args=None, kwargs=None, prefix=None, current_app=None,
...
-                            args=host_args,
-                            kwargs=host_kwargs)
-    path = reverse_path(viewname,
-                        urlconf=host.urlconf,
-                        args=args or (),
-                        kwargs=kwargs or {},
-                        prefix=prefix,
-                        current_app=current_app)
+    # Try all hosts for 3rd party compatibility.
+    if not (getattr(settings, 'HOST_STICKY', True) or (host or host_args or host_kwargs)):
```
Comment (jezdez) [inline django_hosts/resolvers.py]: Maybe make that attr access not happening at run time?

Comment (jezdez) [inline django_hosts/resolvers.py]: also can you turn this around, the code first checks with not, while it could be turned around easily

**Comment by jezdez** [comment]
If the "looping" feature should be enabled by default HOST_STICKY needs to default to `False` since that implies that the host resolving sticks to the explicitly provided hostname.

**Comment by apollo13** [comment]
Not sure if `HOST_STICKY` is the best name then ;) `HOST_REVERSE_LOOPING` maybe? I know it is long, but should describe the intent better.

**Comment by jezdez** [comment]
@apollo13 That's equally confusing since it could mean that the looping is done in reverse ;)

**Comment by apollo13** [comment]
Looping is done in reverse...

**Comment by apollo13** [comment]
@jezdez read the above as: I'll implement whatever you want, just tell me
what you want ;)

**Comment by jezdez** [comment]
Ok, this needs a few things to be complete:

- A non-confusing setting name that takes current behavior into account (opt-in or opt-out?)
- A plan for deprecation (if needed) for the current behavior
- Raising exceptions if it find ambiguous host names like the ones that take parameters
- A migration guide for the docs with noob-friendly explanation what changes

**Comment by timgraham** [comment]
Closing due to inactivity.

**Pull request closed without merging**